sim/FreeRTOS.cpp

```diff
@@ -1,3 +1,44 @@
 #include "FreeRTOS.h"
+#include <algorithm>
+#include <map>
+#include <stdio.h>
+#include <stdlib.h>
 
 void NVIC_SystemReset(void) {}
+
+void APP_ERROR_HANDLER(int err) {
+  fprintf(stderr, "APP_ERROR_HANDLER: %d", err);
+}
+
+namespace {
+std::map<void *, size_t> allocatedMemory;
```

Comment:
do we actually need an ordered map? I think an `unordered_map` would do just fine and has better runtime for search of `O(1)` in the best case

see: https://stackoverflow.com/a/13799886

```suggestion
std::unordered_map<void *, size_t> allocatedMemory;
```

Reply:
Yes, indeed, we don't need the map to be ordered!